doc/main/tbb_userguide/Advanced_Topic_Other_Kinds_of_Iteration_Spaces.rst

```diff
@@ -72,4 +72,39 @@ along its longest axis. When used with ``parallel_for``, it causes the
 loop to be "recursively blocked" in a way that improves cache usage.
 This nice cache behavior means that using ``parallel_for`` over a
 ``blocked_range2d<T>`` can make a loop run faster than the sequential
-equivalent, even on a single processor.
+equivalent, even on a single processor. 
+
+Also, ``blocked_range2d`` allows to use different value types across
+its first dimension (called "rows") and the second one ("columns").
+That allows combining indexes, pointers, and iterators into a joint
+iteration space. The method functions ``rows()`` and ``cols()`` return
+corresponding dimensions in the form of a ``blocked_range``.
+
+The ``blocked_range3d`` class template extends this approach to 3D by adding
+``pages()`` as the first dimension, followed by ``rows()`` and ``cols()``.
+
+The ``blocked_nd_range<T,N>`` class template represents a blocked iteration
```

Comment:
I'm not sure that I got it right, but maybe smth like:

The ``blocked_nd_range<T,N>`` class template represents a blocked iteration space of any dimensionality. However, unlike other classes, all dimensions in ``blocked_nd_range`` must share the same value type. Instead of specifying boundary values directly, you pass ``N`` instances of ``blocked_range<T>`` to the constructor. This difference is also reflected in its different naming pattern.